[master] Enable executor properties in salt-ssh

m-czernek opened the pull request:
### What does this PR do?

The `salt-ssh` binary does not accept the `module_executors` and `executor_opts` properties, yet the `salt` binary does. This PR unifies the differences so that we can use the executors property for `salt-ssh` as well.

### Previous Behavior
`salt-ssh` could only read the `module_executors` property from the roster file (`minion_opts`) or from the `ssh_minion_opts` config.

### New Behavior
`salt-ssh` can read the `module_executors` property from the command-line arguments as well.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

m-czernek committed:
Enable executor properties in salt-ssh

Akm0d reviewed (dismissed):
LGTM but needs a test

twangboy reviewed (changes requested):
Please add a changelog and a test for this.

dwoz committed:
Add tests and changelog for salt-ssh executor options

Tests verify that module_executors and executor_opts propagate from
master opts into Single.minion_opts, and that SaltSSHOptionParser
exposes the --module-executors and --executor-opts flags.

Closes #66340

twangboy commented on salt/client/ssh/__init__.py:
```diff
@@ -1238,6 +1238,10 @@ def __init__(
+        if opts.get("module_executors") is not None:
+            self.minion_opts.update({"module_executors": opts.get("module_executors")})
+        if opts.get("executor_opts") is not None:
+            self.minion_opts.update({"executor_opts": opts.get("executor_opts")})
```
Switch from .update() to .setdefault() so that the global master/CLI options are only applied as a fallback when they aren't already explicitly declared on the minion:

```python
        # Fall back to global master config/CLI options if not explicitly 
        # defined for this target minion in the roster.
        if opts.get("module_executors") is not None:
            self.minion_opts.setdefault("module_executors", opts.get("module_executors"))
        if opts.get("executor_opts") is not None:
            self.minion_opts.setdefault("executor_opts", opts.get("executor_opts"))
```

Additionally, the new unit test test_executor_opts_propagate_to_minion_opts passes because the test fixture cleanly strips these keys out of the base options beforehand, which masks this clobbering edge case.

Please write an additional test case that ensures an existing module_executors key inside ssh_minion_opts isn't overridden by the master config.

m-czernek commented:
This feature work got priority removed since the PR creation, closing this PR for now.
